util/TomlChecker.js

```diff
@@ -0,0 +1,30 @@
+import * as TOML from '@ltd/j-toml';
+import * as os from 'os';
+import path from 'path';
+import { readFileSync, existsSync } from 'fs';
+
+/**
+ * Checks if the .explainer-config.toml file exists in the user's home directory,
+ * and returns the options in it if valid. Returns an empty object if the file doesn't exist.
+ *
+ * @returns {Record<string, any>} - An object containing attributes from the TOML file, or an empty object if the file doesn't exist.
+ * @throws {Error} - If the file exists but cannot be parsed as a valid TOML file.
+ */
+const TomlChecker = () => {
+    const tomlFile = path.join(os.homedir(), '.explainer-config.toml');
+
+    if (!existsSync(tomlFile)) {
+        return {};
+    }
+
+    try {
+        const finalPath = path.resolve(tomlFile);
+        const fileContents = readFileSync(finalPath, 'utf8');
+        return TOML.parse(fileContents);
+    } catch (err) {
+        console.error('Error parsing the TOML file:', err);
+        process.exit(1);

```

Comment:
not sure why this is being done this way. The program has only one entry and exit in `index.js`. You can just do a throw here. Follow https://github.com/aamfahim/explainer.js/blob/29c31bd8ab656243296f7370bf41014c40ca75fe/util/processFileWithProvider.js#L27

The error will be caught in the `try....catch` block in `index.js`.

Reply:
The entire try catch block was removed. Not quite sure why. The try catch should still be there (since TOML can still throw) but should just throw instead, as shown in the referenced file. The error should include message similar to this
> TOML parsing failed: + error.message

Reply:
Ah you're right, need to throw the correct error message, updated

Reply:
also for consistency use `error` instead of `err`

Reply:
updated `err` to `error`

Reply:
should be `error.message` in the thrown error as well, as mentioned in the pervious comment.

Reply:
Updated the way the error message is structured in code